src/model/submissions.nit

```diff
@@ -209,6 +221,60 @@ q
 	end
 end
 
+redef class MissionStar
+	# Check if the star is unlocked for the `program`
+	# Also update `status`
+	fun check(program: Program, status: MissionStatus): Bool do return false
+end
+
+redef class ScoreStar
+	redef fun check(program, status) do
+		var score = self.score(program)
+		if score == null then return false
+
+		# Search or create the corresponding StarStatus
+		# Just iterate the array
+		var star_status = null
+		for ss in status.star_status do
+			if ss.star == self then
+				star_status = ss
+				break
+			end
+		end
+		if star_status == null then
+			star_status = new StarStatus(self)
+			status.star_status.add star_status
+		end
+
+		# Best score?
+		var best = star_status.best_score
+		if best == null or score < best then
+			star_status.best_score = score
+			if best != null then print "STAR new best score {title}. {score} < {best}"
```

Comment:
From memory

~~~nit
class StarNotification
	super PlayerNotification
	serialize
	autoinit(start_status)

	redef var player is lazy do return start_status.player

	var start_status: StarStatus

	redef var object = "Star unlocked"
	redef var body = "New best score {star_status.star.score} for {star_status.star.title}."
	redef var icon = "star"
end
~~~

and then 

~~~nit
player.add_notification(config, new StarNotification(starstatus))
~~~